homeassistant/components/climate/generic_thermostat.py

```diff
@@ -217,4 +217,4 @@ def _control_heating(self):
     @property
     def _is_device_active(self):
         """If the toggleable device is currently active."""
-        return switch.is_on(self.hass, self.heater_entity_id)
+        return switch.async_is_on(self.hass, self.heater_entity_id)
```

Comment:
I think that should not by async since `track_state_change(hass, sensor_entity_id, self._sensor_changed)` don't execute `self._sensor_changed` in async loop while he not have a `@asyncio.coroutine` declerator. This function also call by setup thread.